server/oauth-core/src/main/java/org/eclipse/lyo/server/oauth/core/OAuthRequest.java

```diff
@@ -122,12 +126,16 @@ public LyoOAuthConsumer getConsumer() {
      * @throws OAuthException if the request fails validation
      */
     public void validate() throws OAuthException, IOException, ServletException {
+        log.trace("validating the request.");
         try {
             OAuthConfiguration config = OAuthConfiguration.getInstance();
             config.getValidator().validateMessage(message, accessor);
             config.getTokenStrategy().validateAccessToken(this);
         } catch (URISyntaxException e) {
             throw new ServletException(e);
+        } catch (OAuthProblemException e) {
+            log.error("OAuthProblemException caught when validating the request. {}", e.toString());
```

Comment:
Two questions:

1. Are you sure the exception will only be raised if ther is a problem with our code? If a bad request could cause it too, I'd suggest a `.warn` level. In the case of a production deployment / DDoS attack, you may want to log errors. And this should not include user-caused errors.
2. Could you please paste a typical log line? Are you reasonably sure that `OAuthProblemException` and all subclasses would not print any confidential data?

Reply:
1. Here are the high-level checks being done. So, this exception is not about problems with our code. It's more when the client's signature/token/keys do not match the server - for whatever reason.  So I take it you suggest "warn" here?
```
validateVersion(message);
validateTimestampAndNonce(message);
validateSignature(message, accessor);
oAuthRequest.getConsumer().consumerKey.equals(actualValue)

```

2. Here's the logic behind e.toString() https://github.com/spullara/oauth/blob/1c19b9fd71764646acdd642c314d10a975398a67/commons/src/main/java/net/oauth/OAuthProblemException.java#L114
Note that the code was (and still does) throw this OAuthProblemException (with any potential sensitive data). so, we cannot be doing things worst? (and I believe the exception does not hold sensitive data).

Reply:
I believe the log informatiion is safe.